### PR TITLE
PLT-8038: Adding a copy link option for the buttons in the desktop app for sidebar channels

### PR DESCRIPTION
**Pull request by MusikPolice**
#### Summary
As a part of [PLT-7773](https://mattermost.atlassian.net/browse/PLT-7773), we replaced the LHS channel links with buttons in the Desktop app. The goal of this change was to avoid rendering the URL popover when users hovered the mouse over an LHS channel link, because the popover obscured portions of the view and is unnecessary for internal Mattermost functionality.

Unfortunately, when we made this change, we also lost the ability to right-click on a channel link and copy the channel URL to the user's clipboard. This PR adds a right-click context menu to the LHS channel links when the user is on the Desktop app. This context menu gives the user the option to copy the channel link to their clipboard. It is modeled on the context menu that Electron pops when the user right-clicks on a url anywhere else in the desktop application.

**Note:** This functionality is _only_ available on the desktop app. It has no impact on the webapp, so to test it, you must connect to SpinMint from the desktop application.

screen shot 2018-01-04 at 3 47 28 pm

#### Caveats:
The CSS is still a little bit wrong in the Desktop app. I'll try to sort this out, but it could probably use a review from Asaad.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-8038

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates

### CODE REVIEW

File: components/sidebar.jsx [hunk 1]
```diff
@@ -685,22 +679,58 @@ export default class Sidebar extends React.Component {
...
-                >
-                    {icon}
-                    <span className='sidebar-item__name'>{displayName}</span>
-                    {badge}
-                    {closeButton}
-                </button>
+                <span>
+                    {contextMenu}
+                    {contextMenuTrigger}
+                </span>
```
Comment (MusikPolice) [inline components/sidebar.jsx]: I suspect that the addition of this `<span />` is what's causing the CSS dissimilarities in the LHS between the webapp UI and the desktop UI. I'm trying to track down the CSS selector that is getting confused by this.

Comment (jasonblais) [reply components/sidebar.jsx]: @MusikPolice We can wait for Asaad to return next week and have him take a look. 

In addition to styling, can we also match the functionality of the webapp? I.e. 
 - name = `Copy Link Address`
 - clicking the option copies the link to the clipboard

image

Comment (MusikPolice) [reply components/sidebar.jsx]: I can change the text in the popover to be "Copy Link Address" to match the text in browser context menus. I do think, however, that we should continue to pop the modal to copy the link out of, as that matches what we do for other links, such as the team invite link.

Comment (jasonblais) [reply components/sidebar.jsx]: @MusikPolice Why deviate from webapp experience?

Comment (MusikPolice) [reply components/sidebar.jsx]: @jasonblais removed the modal in https://github.com/mattermost/mattermost-webapp/pull/547/commits/a1426ab74800e6fcc0144606fe4023f82c58392f

**Comment by asaadmahmood** [comment]
@jasonblais Submitted the css fix.

**Comment by MusikPolice** [comment]
Thanks @asaadmahmood

**Comment by jasonblais** [comment]
@MusikPolice @asaadmahmood Looks like there's a build failure. Please help resolve it and I'll generate a new spinmint. Thanks!

**Comment by MusikPolice** [comment]
@jasonblais it just needed to be rebased. Should build now.

**Comment by jasonblais** [comment]
@MusikPolice The ticket mentions

> Adding a copy link option for buttons in the desktop app channels and teams in the left-hand sidebar.

However, this is for channels only. Shall I create a separate one for team icons?

---- 

Tested on Mac, and functionality and CSS looks good for the channels.

**Comment by jasonblais** [comment]
Thanks @MusikPolice! Looks like the build is failing though, might just need a rebase.

**Comment by MusikPolice** [comment]
@jasonblais that was expected. There was a test failure. @jespino helped me sort it out. 

Everything should be done now, but @asaadmahmood will need to take a pass - I can't figure out why the LHS is occluding the context menu on teams. Tried playing with z-index, but nothing seemed to work.

**Pull request merged**